### PR TITLE
Re-factored and fixed some code

### PR DESCRIPTION
**Pull request by vidurangaw**
Signed-off-by: Viduranga Wijesooriya vpowerrc@gmail.com

### CODE REVIEW

File: config/db_config.yml.example [hunk 1]
```diff
@@ -1,13 +1,13 @@
-session_secret: <%= ENV["SYSOPIA_SESSION_SECRET"] %>
-timezone_offset: <%= ENV["SYSOPIA_TIMEZONE_OFFSET"] %>
+session_secret: changeme
+timezone_offset: 0
 database:
   common: &common
     adapter: mysql2
     encoding: utf8
     reconnect: true
-    host: <%= ENV["SYSOPIA_DB_HOST"] %>
```
Comment (dimus) [inline config/db_config.yml.example:8]: I understand why did you make this change, as it is a nuisance to add env file for every command. Howerver we use Docker to deploy Sysopia for EOL system (you can find Dockerfile in production branch). This was the reason behind public config file with 'secret' ENV variables. It is also a pattern that becomes used often in RoR community -- public configs with private ENV vars

May be we should make use of config/env.sh file by default and have settings that make sense for most dev enviromnets. For tests we would just need to change SYSOPIA_ENV to test in spec_helper, so reading environments for a private file would only be needed in production

File: config/environment.rb [hunk 2]
```diff
@@ -25,17 +26,7 @@ def self.db_connection
   end
 
   def self.read_env
-    if ENV["SYSOPIA"]
```
Comment (dimus) [inline config/environment.rb:28]: This method allows to keep environments "in sync" so if someone misses a variable in their working file it becomes obvious.

File: lib/sysopia.rb [hunk 3]
```diff
@@ -1,4 +1,3 @@
-require_relative "../environment"
```
Comment (dimus) [inline lib/sysopia.rb:1]: the intent of getting environment loaded here is decoupling of models and library from presentation layer (helpers, 'controllers'  and views. However as this project is practically all about presentation -- I think it is ok to couple them as you did, so I am ok with the change

File: config/environment.rb [hunk 4]
```diff
@@ -44,13 +35,30 @@ def self.check_env
     extra = e_real - e_required
     raise("Missing env variables: #{missing.join(', ')}") unless missing.empty?
     raise("Extra env variables: #{extra.join(', ')}") unless extra.empty?
+
+    return {
+      "database" => { env.to_s => { 'host' => ENV["SYSOPIA_DB_HOST"], 'username' => ENV["SYSOPIA_DB_USERNAME"], 'password' => ENV["SYSOPIA_DB_PASSWORD"] } },
```
Comment (dimus) [inline config/environment.rb:40]: I would recommend to follow [Ruby style guide](https://github.com/bbatsov/ruby-style-guide) and keep lines under 80 columns

File: .gitignore [hunk 5]
```diff
@@ -10,6 +10,7 @@
 /test/version_tmp/
 /tmp/
 .sass-cache
+config/db_config.yml
```
Comment (dimus) [inline .gitignore:13]: There is a reason we do not have a private config, it is explained below

File: config/db_config.yml.example (hunk 1 above)
Comment (vidurangaw) [inline config/db_config.yml.example:8]: According my changes we can still use ENV variables. Refer https://github.com/EOL/sysopia/blob/de0e020512b64de4999955bf2bb2abfe583e1362/config/environment.rb#L52
config.yml will be only used if ENV variables are not set.

**Pull request closed without merging**